state/indexer/block/kv/kv.go

```diff
@@ -95,11 +100,27 @@ func (idx *BlockerIndexer) Search(ctx context.Context, q *query.Query) ([]int64,
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse query conditions: %w", err)
 	}
+	// conditions to skip because they're handled before "everything else"
+	skipIndexes := make([]int, 0)
+
+	// Check whether we want to return heights where the conditions are true
+	// within the same event or it does not matter
+	matchEvents, matchEventIdx := lookForMatchEvent(conditions)
 
+	if matchEventIdx != -1 {
+		skipIndexes = append(skipIndexes, matchEventIdx)
+	}
```

Comment:
What happens in this logic when someone submits a query like: `match.events=1 AND match.events=1 AND match.events=1`?

Reply:
You are right, same as @sergio-mena  [above](https://github.com/tendermint/tendermint/pull/9759#discussion_r1044869521). We should sanitize the query string and remove any occurences of `match.events`.  Actually it is very easy to ignore their occurrences in the code, but if someone were to enter `"match.events=1"&match_events=false`, the RPC values would be ignored. Will push a fix shortly.